src/sagemaker_training/mpi.py

```diff
@@ -95,18 +95,20 @@ def run(
 
         if wait:
             logger.info("Waiting for MPI process to finish.")
-            _wait_orted_process_to_finish()
+            self._wait_master_to_finish()
             logger.info("Orted process exited")
-            time.sleep(30)
         logger.info("MPI process finished.")
 
     def _wait_master_to_start(self):  # type: () -> None
         while not _can_connect(self._master_hostname):
             time.sleep(1)
 
     def _wait_master_to_finish(self):  # type: () -> None
+        logger.info("Check master node connectivity")
+        sleep_timer = 30
         while _can_connect(self._master_hostname):
-            time.sleep(30)
+            logger.info(f"Can connect to master node. Sleeping for {sleep_timer} seconds")
```

Comment:
Can we make this logging optional and by default `False`? Log can become messy, especially for the notebook instance user. When user submit a training job from notebook instance it will propagate every node's log into the notebook console.